File: app/src/main/java/me/devsaki/hentoid/fragments/intro/SourcesIntroFragment.java
package me.devsaki.hentoid.fragments.intro;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.mikepenz.fastadapter.FastAdapter;
import com.mikepenz.fastadapter.adapters.ItemAdapter;

import java.util.ArrayList;
import java.util.List;

import me.devsaki.hentoid.R;
import me.devsaki.hentoid.enums.Site;
import me.devsaki.hentoid.viewholders.SiteItem;

public class SourcesIntroFragment extends Fragment {

    private final ItemAdapter<SiteItem> itemAdapter = new ItemAdapter<>();

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.intro_slide_06, container, false);

        // Recycler
        List<SiteItem> items = new ArrayList<>();
        for (Site s : Site.values())
            if (s.isVisible()) items.add(new SiteItem(s, true, false));
        itemAdapter.add(items);

        FastAdapter<SiteItem> fastAdapter = FastAdapter.with(itemAdapter);
        RecyclerView recyclerView = view.findViewById(R.id.list);
        recyclerView.setAdapter(fastAdapter);

        return view;
    }

    public List<Site> getSelection() {
        List<Site> result = new ArrayList<>();
        for (SiteItem s : itemAdapter.getAdapterItems())
            if (s.isSelected()) result.add(s.getSite());
        return result;
    }
}

File: app/src/main/java/me/devsaki/hentoid/fragments/intro/SourcesIntroFragment.kt
package me.devsaki.hentoid.fragments.intro

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.mikepenz.fastadapter.FastAdapter
import com.mikepenz.fastadapter.adapters.ItemAdapter
import me.devsaki.hentoid.R
import me.devsaki.hentoid.databinding.IntroSlide06Binding
import me.devsaki.hentoid.enums.Site
import me.devsaki.hentoid.viewholders.SiteItem

class SourcesIntroFragment : Fragment(R.layout.intro_slide_06) {
    private var binding: IntroSlide06Binding? = null

    private val itemAdapter = ItemAdapter<SiteItem>()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = IntroSlide06Binding.inflate(inflater, container, false)

        // Recycler
        val items: MutableList<SiteItem> = ArrayList()
        for (s in Site.entries.sortedBy { s -> s.description })
            if (s.isVisible) items.add(
                SiteItem(s, selected = true, showHandle = false)
            )
        itemAdapter.set(items)

        val fastAdapter: FastAdapter<SiteItem> = FastAdapter.with(itemAdapter)
        binding?.list?.adapter = fastAdapter

        return binding?.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }

    fun getSelection(): List<Site> {
        val result: MutableList<Site> = ArrayList()
        for (s in itemAdapter.adapterItems) if (s.isSelected) result.add(s.site)
        return result
    }
}